stdlib/public/SwiftShims/swift/shims/RefCount.h

```diff
@@ -397,7 +397,7 @@ class RefCountBitsT {
   bool isOverflowingUnownedRefCount(uint32_t oldValue, uint32_t inc) const {
     auto newValue = getUnownedRefCount();
     return newValue != oldValue + inc ||
-      newValue == Offsets::UnownedRefCountMask;
+      newValue == Offsets::UnownedRefCountMask >> Offsets::UnownedRefCountShift;
```

Comment:
I think we'd do better using bitfields instead of a bunch of constants and shifts, then we wouldn't need masks and could just deal in a plain "max value." But that would be a major overhaul. Short of that, I can't think of anything except for better test coverage. Hopefully we'll be in a bit better shape in that respect now.

Reply:
It might be acceptable with a union of multiple different bitfield layouts. We'd need one with `strong: 22; useSlowRC: 1` and one with `strongAndUseSlowRC: 23`, for example. That might end up just being worse.

Reply:
That'd be UB these days though; you'd have to add some `memcpy()`s, which hopefully would get optimised away, but still.

We could also have done a "fake bitfield" thing with operator overloads and special types that actually do the relevant shifting and masking under the covers. I didn't think that was much better either :-)